protocol.md

```diff
@@ -114,6 +114,114 @@ Observe the behavior of this algorithm in some failure scenarios. First, any of
+
+During the original pairing, it was possible to send a hash of a document such as a PDF file that described how the sharer would like the helper to respond during recovery. It might ask that during any recovery, the helper first do the pairing, and then spend a week trying to contact the owner of the old secret (by text, email, and paper mail), to let the sharer know that someone is trying to do a recovery in their name. This would be especially important if the authentication is not done in person. Or, it could request that  the authentication be done in person, and might even involve biometrics, so it would be difficult for an attacker to impersonate the sharer. But even with that more secure authentication, the sharer might still request that the helper spend a week contacting them before recovery. Or they might not. It is up to the app and the user.
+
+Once the helper is ready to help with recovery (perhaps after that one-week wait), the helper will then respond to requests for the share by sending the share. After the sharer receives each new share from each new pairing, they will try to reconstruct the secret. When they finally have enough pairings, they will discover that the reconstruction is successful. At that point, they have successfully recovered the old secret. The old secret should contain the contact info and public keys of all the helpers, as well as the private keys of the sharer. So once that old secret is recovered, it will be possible for the sharer to use that old secret ID and keys to again contact all the old helpers.
+
+At this point, the sharer\'s app has a choice, depending on how it\'s configured, or possibly by asking the user what to do. It can send requests to unpair the new secret ID from all the helpers, and go back to using the old secret ID as before. Or, it could copy all the information from the old secret to the new secret, and send unpair requests to all the old helpers to remove the old secret ID. Either way, if it had allowed the user to add new info to the new secret, then it could combine the info from both old and new secret, and put it all into whichever secret it is keeping. And then unpair and discard the other secret ID. One advantage of deleting the new secret, is that allows all the old helpers to continue helping, even those who didn\'t re-pair during recovery. One advantage of deleting the old secret is that it allows moving to the new keys, and so is equivalent to doing a key rotation after losing the phone with the old keys. So that may seem to be better for security. The app can choose either way, and the protocol supports either approach.
+
+## Authentication
+
+It is up to each app to decide how the helper and user authenticate each other. The authentication is outside the protocol itself, and it happens before the first message of the protocol is sent. Here are three example scenarios of how authentication might be done, though other methods are possible.
```

Comment:
The purpose of QR codes generated on the fly with a nonce is to avoid MITM attacks.  Other approaches to authentication will need to have equally good protection against MITM attacks. Either way, it's outside the protocol. But it's not outside what the DeRec Alliance cares about. We should definitely create documents about what we believe are best practices.